src/utils.ts

```diff
@@ -46,17 +52,26 @@ class ReactElementTypeError extends Error {
   }
 }
 
-function checkReactElement(element, ...args) {
-  if (!VALID_ELEMENTS.includes(element.type) && !element._fiber) {
-    throw new ReactElementTypeError(element, ...args);
+function checkReactElement(
+  element: ReactTestInstance | null,
+  matcherFn: Function,
```

Comment:
Maybe `jest.CustomMatcher` would do? https://github.com/DefinitelyTyped/DefinitelyTyped/blob/bf0eabf1a39b3eefe344695b27694677cc177a70/types/jest/index.d.ts#L567

Reply:
Changed it